src/libais/ais8_367.cpp

```diff
@@ -274,5 +274,459 @@ ostream& operator<< (ostream &o, const Ais8_367_25 &msg) {
   return o << msg.mmsi << ": " << msg.version << ": " << msg.utc_hour << ": " << msg.utc_min << ": " << msg.pressure << ": " << msg.wind_speed << ": " << msg.wind_dir;
 }
 
+// Environmental Message
+Ais8_367_33_Location::Ais8_367_33_Location(const AisBitset &bits,
+                                           const size_t offset) {
+  version = bits.ToUnsignedInt(offset, 6);
+  position = bits.ToAisPoint(offset + 6, 55);
+
+  precision = bits.ToUnsignedInt(offset + 61, 3);
+
+  altitude_raw = bits.ToUnsignedInt(offset + 64, 12);
+  altitude = altitude_raw / 10.;  // meters
+
+  owner = bits.ToUnsignedInt(offset + 76, 4);
+  timeout = bits.ToUnsignedInt(offset + 80, 3);
+  spare2 = bits.ToUnsignedInt(offset + 83, 2);
+}
+
+Ais8_367_33_Station::Ais8_367_33_Station(const AisBitset &bits,
+                                         const size_t offset) {
+  name = bits.ToString(offset, 84);
+  // Remove any trailing '@' characters.  Spec says they are not to be shown in the presentation.
+  while (name.back() == '@') {
+    name.pop_back();
+  }
+  spare2 = bits.ToUnsignedInt(offset + 84, 1);
+}
+
+Ais8_367_33_Wind::Ais8_367_33_Wind(const AisBitset &bits,
+                                   const size_t offset) {
+  wind_speed = bits.ToUnsignedInt(offset, 7);
+  wind_gust  = bits.ToUnsignedInt(offset + 7, 7);  // knots
+  wind_dir = bits.ToUnsignedInt(offset + 14, 9);
+  wind_gust_dir = bits.ToUnsignedInt(offset + 23, 9);
+  sensor_type = bits.ToUnsignedInt(offset + 32, 3);
+  wind_forecast = bits.ToUnsignedInt(offset + 35, 7);
+  wind_gust_forecast = bits.ToUnsignedInt(offset + 42, 7);  // knots
+  wind_dir_forecast = bits.ToUnsignedInt(offset + 49, 9);
+  utc_day_forecast = bits.ToUnsignedInt(offset + 58, 5);
+  utc_hour_forecast = bits.ToUnsignedInt(offset + 63, 5);
+  utc_min_forecast = bits.ToUnsignedInt(offset + 68, 6);
+  duration = bits.ToUnsignedInt(offset + 74, 8);
+  spare2 = bits.ToUnsignedInt(offset + 82, 3);
+}
+
+Ais8_367_33_WaterLevel::Ais8_367_33_WaterLevel(const AisBitset &bits,
+                                               const size_t offset) {
+  type = bits[offset];
+  level = bits.ToInt(offset + 1, 16);
+  trend = bits.ToUnsignedInt(offset + 17, 2);
+  vdatum = bits.ToUnsignedInt(offset + 19, 5);
+  sensor_type = bits.ToUnsignedInt(offset + 24, 3);
+  forecast_type = bits[offset + 27];
+  level_forecast = bits.ToInt(offset + 28, 16);
+  utc_day_forecast = bits.ToUnsignedInt(offset + 44, 5);
+  utc_hour_forecast = bits.ToUnsignedInt(offset + 49, 5);
+  utc_min_forecast = bits.ToUnsignedInt(offset + 54, 6);
+  duration = bits.ToUnsignedInt(offset + 60, 8);
+  spare2 = bits.ToUnsignedInt(offset + 68, 17);
+}
+
+Ais8_367_33_Curr2D::Ais8_367_33_Curr2D(const AisBitset &bits,
+                                       const size_t offset) {
+  for (size_t idx = 0; idx < 3; idx++) {
+    size_t start = offset + idx * 26;
```

Comment:
`const size_t start`  Better to be aggressive with adding const here and elsewhere.  It doesn't really change anything for the compiler, but it tells the reader that there aren't going to be any hidden surprises like `++`